core/src/main/java/com/microsoft/applicationinsights/internal/util/PropertyHelper.java

```diff
@@ -78,18 +84,75 @@ public static Properties getSdkVersionProperties() {
     /**
      * A method that loads the properties file that contains the AI SpringBootStarter version number
      * @return The properties or null if not found.
+     * @deprecated Use {@link #getStarterVersionString()} to get the cached starter version value.
      */
+    @Deprecated
     public static Properties getStarterVersionProperties() {
         try {
             return getProperties(STARTER_VERSION_FILE_NAME);
         }
         catch (IOException e) {
-            InternalLogger.INSTANCE.trace("Could not find starter version file: %s,"
-                + "stack trace is: ", SDK_VERSION_FILE_NAME, ExceptionUtils.getStackTrace(e));
+            InternalLogger.INSTANCE.error("Could not find starter version file '%s': %s", SDK_VERSION_FILE_NAME, ExceptionUtils.getStackTrace(e));
         }
         return null;
     }
 
+    /**
+     * Returns the SDK version string, the <i>version-number</i> prefixed with "java:".
+     * The <i>version-number</i> is the value of the {@value #VERSION_PROPERTY_NAME} property in {@value #SDK_VERSION_FILE_NAME}.
+     * If the properties file cannot be read, {@value #UNKNOWN_VERSION_VALUE} is used for the <i>version-number</i>.
+     *
+     * @return "java:<i>version-number</i>" or "java:unknown"
+     */
+    public static String getSdkVersionString() {
```

Comment:
would it make sense to reaname method as `getSdkVersionAsString()` to be a bit more clear?

Reply:
sure